core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -1167,6 +1175,141 @@ def load(cls, path: str, reset_paths: bool = True, verbose: bool = True):
+
+        Returns
+        -------
+        path : str
+            Path to the saved curves, minus the file name.
+        """
+        if path is None:
+            path = self.path
+        if type(metrics) == str:
+            metrics = [metrics]
+        if len(metrics) == 0:
+            raise ValueError("At least one metric must be specified to save generated learning curves.")
+
+        os.makedirs(path, exist_ok=True)
+        file_path = os.path.join(path, self.learning_curve_file_name)
+        out = self._make_learning_curves(metrics=metrics, curves=curves)
+
+        with open(file_path, "w") as json_file:
+            json.dump(out, json_file, indent=4)
+
+        return file_path
+
+    def _make_learning_curves(self, metrics: str | List[str], curves: dict[dict[str : List[float]]]) -> List[List[str], List[str], List[List[float]]]:
+        """
+        Parameters
+        ----------
+        metrics : str or list(str)
+            List of all evaluation metrics computed at each iteration of the curve
+        curves : dict[dict[str : list[float]]]
+            Dictionary of evaluation sets and their learning curve dictionaries.
+            Each learning curve dictionary contains evaluation metrics computed at each iteration.
+            See Abstract Model's save_learning_curves method for a sample curves input.
+
+        Returns
+        -------
+        List[List[str], List[str], List[List[float]]]: The generated learning curve artifact.
+            if eval set names includes: train, val, or test
+            these sets will be placed first in the above order.
+        """
+
+        # ensure main eval sets first: train, val, test
+        items = []
+        order = ["train", "val", "test"]
+        for eval_set in order:
+            if eval_set in curves:
+                items.append((eval_set, curves[eval_set]))
+                del curves[eval_set]
+
+        items.extend(curves.items())
+        eval_sets, curves = list(zip(*items))
+
+        data = []
+        for metric in metrics:
+            data.append([c[metric] for c in curves])
+
+        return [eval_sets, metrics, data]
+
+    @classmethod
+    def load_learning_curves(cls, path: str) -> List:
+        """
+        Loads the learning_curve data from disk to memory.
+
+        Parameters
+        ----------
+        path : str
+            Path to the saved model, minus the file name.
+            This should generally be a directory path ending with a '/' character (or appropriate path separator value depending on OS).
+            The model file is typically located in os.path.join(path, cls.model_file_name).
+
+        Returns
+        -------
+        learning_curves : List
+            Loaded learning curve data.
+        """
+        if not cls._get_class_tags().get("supports_learning_curves", False):
+            raise AssertionError("Attempted to load learning curves from model without learning curve support")
+
```

Comment:
```python
if not self.saved_learning_curves:
    raise AssertionError(...)
```

Reply:
This is in a classmethod, so no access to self

Reply:
Sounds good, that makes sense